pkg/controller/knicluster/knicluster_controller.go

```diff
@@ -251,6 +278,28 @@ func (r *ReconcileKNICluster) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
+	// Add conditions if there are none
+	if instance.Status.Conditions == nil {
+		status.SetStatusCondition(&instance.Status.Conditions, kniv1alpha1.Condition{
+			Type:   kniv1alpha1.ConditionAvailable,
+			Status: corev1.ConditionFalse,
+		})
+		status.SetStatusCondition(&instance.Status.Conditions, kniv1alpha1.Condition{
+			Type:   kniv1alpha1.ConditionProgressing,
+			Status: corev1.ConditionTrue,
+		})
+		status.SetStatusCondition(&instance.Status.Conditions, kniv1alpha1.Condition{
+			Type:   kniv1alpha1.ConditionDegraded,
+			Status: corev1.ConditionFalse,
+		})
+		status.SetStatusCondition(&instance.Status.Conditions, kniv1alpha1.Condition{
+			Type:   kniv1alpha1.ConditionUpgradeable,
+			Status: corev1.ConditionUnknown,
+		})
+
+		_ = r.client.Status().Update(context.TODO(), instance)
```

Comment:
I'm not totally sure what to do in the case of an err here. Do we fail the reconcile and requeue?